### PR TITLE
Add validation functions

### PR DESCRIPTION
**Pull request by emanguy**
* Add "0x on the front" validation
 * Add "has only hex characters" validation

### CODE REVIEW

File: src/js/validation/has0x/validateHas0x.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+var AMPParse = AMPParse || {};
+
+AMPParse.validateHas0x = function(hexString)
+{
+    return /^0[xX]/.test(hexString);
+}
```
Comment (gti28) [inline src/js/validation/has0x/validateHas0x.js]: I think in the requirements we said the prefix is optional

**Review by gti28 — CHANGES_REQUESTED** [review]

File: src/js/validation/has0x/validateHas0x.js (hunk 1 above)
Comment (emanguy) [reply src/js/validation/has0x/validateHas0x.js]: If that's the case I can change it

Comment (gti28) [reply src/js/validation/has0x/validateHas0x.js]: looks like it was in the design, not the requirements, so I'm fine with making 0x required, we can update the design to match

**Review by gti28 — APPROVED** [review]

File: src/js/validation/has0x/validateHas0x.js (hunk 1 above)
Comment (emanguy) [reply src/js/validation/has0x/validateHas0x.js]: I'll just remove the 0x validation

**Pull request merged**